aten/src/ATen/FunctionalInverses.cpp

```diff
@@ -128,8 +128,8 @@ Tensor FunctionalInverses::_neg_view_copy_inverse(const Tensor& base, const Tens
 }
 
 Tensor FunctionalInverses::as_strided_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views, at::IntArrayRef size, at::IntArrayRef stride, c10::optional<int64_t> storage_offset) {
-    TORCH_INTERNAL_ASSERT(false, "as_strided has not been implemented in the functionalization pass yet");
-    return Tensor();
+    // Pessimism: we can't reapply views for as_strided_scatter.
```

Comment:
Not sure what that means?

Reply:
I have a similar comment for all of the other "slicing" ops like slice/select/diagonal.

For most operators, the "inverse" can be a view (e.g. the inverse of `permute` is another `permute` call).

`functionalize()` will also by default re-apply views instead of using view_copy operators, so in most of the functions in this file you'll see code like "if (re-apply views) return view() else view_copy()". But for the ops listed above we can't do that - running the "inverse" forces us to allocate a new tensor.